velox/exec/Window.cpp

```diff
@@ -205,6 +246,61 @@ void Window::createPeerAndFrameBuffers() {
     frameStartBuffers_.push_back(frameStartBuffer);
     frameEndBuffers_.push_back(frameEndBuffer);
   }
+
+  // Update peer group bounds for all partitions.
```

Comment:
This shouldn't be done at the level of all data. It could get very large. The peer values array should only be constructed at a per-partition level. Please change your code with that in mind.

Reply:
Thanks for the suggestion, updated accordingly!